src/d3a/models/area/stats.py

```diff
@@ -90,3 +95,39 @@ def cheapest_offers(self):
         for market in self._markets.markets.values():
             cheapest_offers.extend(market.sorted_offers[0:1])
         return cheapest_offers
+
+    def min_max_avg_rate_market(self, time_slot):
+        out_dict = {"min_trade_rate": None,
+                    "max_trade_rate": None,
+                    "avg_trade_rate": None,
+                    "total_traded_energy_kWh": None}
+        for market in self._markets.all_spot_markets:
+            if market.time_slot == time_slot and len(market.trades) > 0:
+                trade_volumes = [trade.offer.energy for trade in market.trades]
+                trade_rates = [trade.offer.price/trade.offer.energy for trade in market.trades]
+                out_dict["min_trade_rate"] = limit_float_precision(min(trade_rates))
+                out_dict["max_trade_rate"] = limit_float_precision(max(trade_rates))
+                out_dict["avg_trade_rate"] = limit_float_precision(mean(trade_rates))
+                out_dict["total_traded_energy_kWh"] = limit_float_precision(sum(trade_volumes))
+        return out_dict
+
+    @property
+    def current_market(self):
```

Comment:
It is a bummer that this property is on the area, and we have to duplicate it here. It's ok for now though.

Reply:
Yeah, I was thinking about this a lot, but i could not come up with a different approach. We could skip the tests in get_market_price_stats and just dont return anything if there is no data, but i thought it might be helpful for the user.